File: Day_1/Plus_Minus/plus_minus.py
#!/bin/python3

import math
import os
import random
import re
import sys

#
# Complete the 'plusMinus' function below.
#
# The function accepts INTEGER_ARRAY arr as parameter.
#

def plusMinus(arr):
    pos_count = 0
    neg_count = 0
    for elt in arr:
        if elt > 0:
            pos_count += 1
        elif elt < 0:
            neg_count += 1
    print(pos_count/len(arr))
    print(neg_count/len(arr))
    print((len(arr)-(pos_count + neg_count))/len(arr))

if __name__ == '__main__':
    n = int(input().strip())

    arr = list(map(int, input().rstrip().split()))

    plusMinus(arr)

File: Day_1/Plus_Minus/plusMinus.ts
/*
 * Complete the 'plusMinus' function below.
 *
 * The function accepts INTEGER_ARRAY arr as parameter.
 */

function plusMinus(arr: number[]): void {
    const total = arr.length
    let plus = 0
    let minus = 0
    let zero = 0
    arr.map((num : number) => {
        if(typeof num === "number"){
            (num > 0 ) ? plus++ : (num < 0) ? minus++ : zero++
        }    
    })
    const proba_plus = plus / total
    const proba_minus = minus / total
    const proba_zero = zero / total
    
    console.log(`${proba_plus}\n${proba_minus}\n${proba_zero}`)

}
